src/androidTest/java/tests/database/GenericDaoTest.java

```diff
@@ -0,0 +1,28 @@
+package tests.database;
+
+import android.content.Context;
+
+import androidx.test.platform.app.InstrumentationRegistry;
+
+import org.junit.After;
+import org.junit.Before;
+
+import io.split.android.client.storage.db.SplitRoomDatabase;
+
+public abstract class GenericDaoTest {
+
+    protected SplitRoomDatabase mRoomDb;
+    protected Context mContext;
+
+    @Before
+    public void setup() {
+        mContext = InstrumentationRegistry.getInstrumentation().getContext();
+        mRoomDb = SplitRoomDatabase.getDatabase(mContext, "encrypted_api_key");
```

Comment:
It would be good to replace the database by the in memory test database here.
```suggestion
        mRoomDb = DatabaseHelper.getTestDatabase(mContext)
```